src/os/bluestore/BlueFS.cc

```diff
@@ -2066,6 +2091,138 @@ int BlueFS::device_migrate_to_new(
   return 0;
 }
 
+int BlueFS::migrate_wal_to_v1() {
+  string wal_dir("db.wal");
+  auto dir_it = nodes.dir_map.find(wal_dir);
+  if (dir_it == nodes.dir_map.end()) {
+    dout(5) << fmt::format("{} No files needed to move", __func__) << dendl;
+    return 0;
+  }
+  
+  DirRef dir = dir_it->second;
+  
+  auto file_it = dir->file_map.begin();
+  size_t number_of_wal_v2 = 0;
+  size_t wal_v2_aggregated_size = 0;
+  std::vector<std::pair<string, FileRef>> files_to_migrate;
+  files_to_migrate.reserve(dir->file_map.size());
+  
+  // get info about wal files
+  while(file_it != dir->file_map.end()) {
+    string file_name = file_it->first;
+    FileRef file = file_it->second;
+    if(file->is_new_wal()) {
+      number_of_wal_v2++;
+      wal_v2_aggregated_size += file->fnode.wal_size;
+      files_to_migrate.push_back({ file_name, file });
+    }
+    file_it++;
+  }
+  
+  // check free space
+  uint8_t wal_bdev = vselector->select_prefer_bdev(vselector->get_hint_by_dir("db.wal"));
```

Comment:
Just a note.
We have WAL_BDEV thing.
For that reason it is better to use "selected_bdev" "pref_bdev" or just bdev.